examples/nextjs-with-contentful/services/services-tools/index.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  createPage,
+  mergePages,
+  RequestCtx,
+  Page,
+  NotFoundError
+} from "@ekino/rendr-core";
+import { Loader } from "@ekino/rendr-loader";
+import { contentfulLoader } from "../contentful-tools";
+
+/**
+ * We create this small helper functions just to avoid code duplication
+ * but allow you to customize it depends on your needs.
+ */
+export async function getPage(page: Page, ctx: RequestCtx) {
+  // retrieve the page from the loader,
+  // as a side note this code work with any loaders, not only contentful ...
+  // as long as the loader return a Page object.
+  const pageFromContentful = await contentfulLoader(ctx, page, () => {});
+
+  // Now we have the page, you can either return the page coming from contentful
+  // return pageFromContentful
+
+  // or merge it with the default one provided by the code
+  return mergePages([page, pageFromContentful as Page]);
+}
+
+// This small loader is used to catch error, so we can update
+// page layout in case of error. The layout for errors should
+// come from a static source - ie, hard coded, to avoid any
+// error coming from the original datasources (datasource not available).
+export const errorBoundaryLoader: Loader = async (ctx, page, next) => {
+  let resultPage: Page | void;
+  try {
+    resultPage = await next();
+  } catch (err) {
+    // something wrong happened
+    console.log("An error occurs while loading page", {
+      message: err.message
+    });
+
+    resultPage = createPage();
+    resultPage.statusCode = err instanceof NotFoundError ? 404 : 500;
+  }
+  return resultPage;
+};
```

Comment:
This can can be part of the loader module by removing the `console.log` reference